### ISSUE TITLE
BIA Review of Architecture Diagram

### ISSUE THREAD

**Issue by gary-fallon** [department-of-veterans-affairs/va.gov-team#53111]
## Description
The Architecture Diagram referenced in the BIA and used in eMASS requires additional information to be added.

- [ ] Determine if additional architecture diagrams can be added to eMASS.

## Tasks
The following tasks must be reviewed and completed as necessary.
- [ ] Update the architecture diagram(s) to include major business processes, minor applications, necessary workflows (e.g., CI/CD), as well as traffic flow, security controls, identified boundaries, and any other relevant information.

## Acceptance
The following tasks must be completed before this issue can be considered done.
- [x] Provide updated diagrams showing: 
   - [x] Vets-Website & Vets-API Application Infrastructure 
   - [x] CMS Application Infrastructure
   - [x] Vets-Website external connections 
- [x] Review the document for accuracy and completeness.

**Comment by little-oddball** [comment]
This ticket is a child of the epic below which has the BIA, etc.:

https://github.com/department-of-veterans-affairs/va.gov-team/issues/53055

**Comment by little-oddball** [comment]
@alyssagallion For diagram updating can you work w/ Jesse House who is the PM and Kyle Matheny?

**Comment by alyssagallion** [comment]
Started a group chat on Slack with @jhouse-solvd and @ph-One

**Comment by alyssagallion** [comment]
Assigning this ticket to @jhouse-solvd per our Slack conversation and meeting today.

**Comment by jhouse-solvd** [comment]
I added my team label (+infrastructure) to show up in our workspace. I'm taking a look at this tonight to review and add some details about what diagrams are needed to satisfy the system description. 

I've got some time set aside tomorrow as well. The plan is to provide @alyssagallion and the team with estimates tomorrow.

**Comment by jhouse-solvd** [comment]
This issue has been rated as a 5 (following the guidance [here](https://vfs.atlassian.net/wiki/spaces/OT/pages/2354642992/Use+points+to+estimate+issue+or+story+complexity#5-%E2%80%93-Longer-time-to-deliver%2C-high-complexity%2C-likely-unknowns)

**Comment by jhouse-solvd** [comment]
_FYSA, I’m OOO tomorrow and Friday (2/23,24) and will work on this upon my return._

We have a good understanding of the scope and details to include. Now we need a few hours to update (or create) diagrams:

- **Update the [BRD architecture](https://vfs.atlassian.net/wiki/spaces/OT/pages/2231304195/Platform+Infrastructure+Diagrams#BRD-Architecture) diagram**
  - Include additional AWS components used by Vets-API and Vets-Website
  - Include identity stuff from the diagram provided by Joe Niquette.
- **Re-create and update the [VSP Topology - VSP Downstream Connection Map](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/d50e7f5d70bb146857672446ad25e3e7398cd4d1/platform/engineering/VSP%20Topology%20-%20VSP%20Downstream%20Connection%20Map.pdf) diagram**
  - Include additional connections found in the forward proxy config
  - Include additional connections found on the Platform Website 
    - External Services Integration
    - Include identity stuff from the diagram provided by Joe Niquette.
- **Create a Remote Access diagram** that shows external connections (SOCKS, DNS, VPCs)
- **Create a CI/CD Diagram** that shows how developers deploy code and applications

Thanks to all who have helped get the info together. If you’re interested, there are [notes and to-dos in my personal Confluence space](https://vfs.atlassian.net/wiki/spaces/~814964285/pages/2538930209/ATO+Work+-+Q1+2023).

**Comment by jhouse-solvd** [comment]
I'm working on this now and plan to have updated diagrams to share by EOD.

**Comment by jhouse-solvd** [comment]
I have an updated diagram showing application (Vets-API) infrastructure and another showing external connections. When Gary has time, I'd like to review with him. 

I'll begin working on the remote access and CI/CD diagrams today.

**Comment by jhouse-solvd** [comment]
Note: This work will carry over into the next sprint.

**Comment by jhouse-solvd** [comment]
I did a review of diagrams w/ Kyle Matheny this morning. He provided great feedback and clarified some technical details that would be good to represent. I'll add notes to this issue soon.

**Comment by jhouse-solvd** [comment]
The external services connections relate to #53885

**Comment by fowusu2** [comment]
@Gary-fallon Is the architecture diagram finalized now

**Comment by jhouse-solvd** [comment]
The external connections diagram has been prioritized due to its relation to #53885. 

I met w/ Joe Niquette and Gary Fallon and have updated the external connections diagram based on their feedback. 

I scheduled time for tomorrow afternoon to make changes needed to provide additional clarity around where various backend services are hosted, e.g.
- within the VAEC? 
- hosted in a DSVA AWS GovCloud account? 
- within the broader VA network? 
- outside of the VAEC, AWS GovCloud, and VA Network?

**Comment by jhouse-solvd** [comment]
I'm working on this now. I plan to have an updated diagram to share in the next couple of hours. 

Please also see the recent comment [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/53885#issuecomment-1462860804) for useful context.

**Comment by jhouse-solvd** [comment]
@gary-fallon - Regarding the updates to the diagrams per yesterday's discussion: 

- I made some changes to reflect Caseflow/appeals as inside of a neighboring VPC
- I've not yet added VA Notify, as I need to dig into the details there a bit better
- I went through each of the services listed in the VA network section on the right, and they all appear to be hosted by the VA, based on the documentation available to me, though I'm not sure of the specifics without further investigation
- I moved some services from the DOD into their own network 
- I moved several services into a third-party data provider section; we could call this something else, but I want to reflect that somehow

I shared the diagram in DSVA Slack to get your feedback. Please let me know if you'd like to discuss this tomorrow or the next.

**Comment by gary-fallon** [comment]
@jhouse-solvd can you please include CMS as part of the AWS architecture diagram? Thank you.

**Comment by jhouse-solvd** [comment]
[X-Post for visibility](https://github.com/department-of-veterans-affairs/va.gov-team/issues/53885#issuecomment-1466855597)

**Comment by jhouse-solvd** [comment]
[X-Post for visibility](https://github.com/department-of-veterans-affairs/va.gov-team/issues/53885#issuecomment-1468893325)

**Comment by gary-fallon** [comment]
@jhouse-solvd can you please share whatever diagrams you are currently working on?

I will be drafting the BIA and ISCP based on a new definition of the FISMA authorization boundary that includes all VFS applications.

I'll unassign this issue for now and follow up if I have questions. Thank you!

**Comment by jhouse-solvd** [comment]
@gary-fallon - Do you still want me to include CMS components on the infrastructure architecture diagram? I met w/ one of their DevOps engineers this morning to inquire about details. 

I'm heading into a Team of Teams meeting right now but I will export and share these diagrams as soon as I'm finished.

**Comment by gary-fallon** [comment]
@jhouse-solvd yes, please finish up whatever you were working on first. Thank you!

**Comment by gary-fallon** [comment]
I reviewed the diagrams provided by @jhouse-solvd.

**Comment by alyssagallion** [comment]
During our Gap Analysis, we realized that the System Security Plan (SSP) was incorrect. We have since updated it, presented it to the ISSO and Ray Wang is now reviewing the associated tickets he needs to approve. Once those are reviewed and approved, @ScottCutlip will continue the work on SSP. 

SSP is the dependency. Until approved, all BIA work is on hold.

**Comment by alyssagallion** [comment]
@jhouse-solvd Can you also add points to this ticket? We are trying to get an accurate Burndown Report. Thanks!

**Comment by jhouse-solvd** [comment]
@alyssagallion - The Epic has 3 stories, each of which are 3 points, making the epic total 9. We could also point the epic, but it will skew the points higher (ie epic + stories). 

Let me know what you'd like to do! :)

**Comment by jhouse-solvd** [comment]
The most recent architecture diagram has been included in the BIA. 

Please [see the parent epic](https://github.com/department-of-veterans-affairs/va.gov-team/issues/53055#issuecomment-1589591963) for future updates. 

Closing.
